cache.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+type Cache interface {
+	read()
+	write()
+	new()
+}
+type CacheBase struct {
```

Comment:
I am not sure what Base will be used here for ? Can you describe more about it's use ?